File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        int mid = r/2;

        while(l <= r){
            if(nums[mid] == target){
                return mid;
            }else if(nums[mid] > target){
                r = mid - 1;
            }else if(nums[mid] < target){
                l = mid + 1;
            }
            mid = (r + l) / 2;
        }

        return -1;
    }
}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        pq = []

        for n in nums:
            heapq.heappush(pq, -n)
        
        popped = 0
        #-5 -4 -3 -2 -1
        for i in range(k):
            popped = -heapq.heappop(pq)
        
        return popped